src/vs/workbench/api/common/extHostTypes.ts

```diff
@@ -2942,14 +2942,14 @@ export class NotebookCellMetadata {
 			change.breakpointMargin ?? this.breakpointMargin,
 			change.runnable ?? this.runnable,
 			change.hasExecutionOrder ?? this.hasExecutionOrder,
-			change.executionOrder ?? this.executionOrder,
+			'executionOrder' in change ? change.executionOrder : this.executionOrder,
```

Comment:
In Jupyter we need to be able to clear this out (setting to `0` still displays the value in the UI, undefined is what we need.

Reply:
Please use `null` for that, so `undefined` means no change and `null` means unset the value. That's how it is done in `Uri.with`. And thanks for looking into this

Reply:
Thanks, will close this issue & the PR.

Reply:
Will do, but this fix is still required isn't it?
I don't see anything that would change the current behaviour even if i used `null`.
Or are you suggesting I use `null` as good practice

Reply:
So `meta.with({executionOrder: undefined, runnable: true })` should be the same as `meta.with({ runnable: true })`. The presence of a property with value `undefined` should be the same as its absence. The uri#with-function uses null or the empty string for erasing/unsetting a value but there it is nice because the empty string is the default for its properties anyways. Not 100% sure how this should be done with executionOrder but lean towards `null` or a dummy value like -1. We should also keep in mind that execution order will be moving out of cell metadata, so some temporary fix might acceptable

Reply:
@DonJayamanne an easy workaround would be to not use `with` but simply create a new instance where the value isn't set

Reply:
I'm assuming we'd reset the other items the same way, create a new object.
Closing the PR & issue.

Reply:
Thanks for the clarification

Reply:
Trying to do that, & its not looking easy.
I cannot do the following:
```typescript
// This is the existing code (doesn't work)
const newMeatadata = cell.metadata.with({executionOrder: undefined});
// Type script doesn't allow this.
const newMeatadata = new NotebookCellMetadat({...cell.metadata, executionOrder: undefined});

// Only solution
const newMeatadata = new NotebookCellMetadat({
	executionOrder: undefined,
	// Set each & every property, manually..
	// This seem like a bad appproach, tomorrow if you add a new property we will have to update this coee & 99% of the time, we'll forget it.
});
```

Possible I'm missing something here, but i cannot find an easy way to unset the `executionOrder`, please could you provide a sample code

Thanks

Reply:
I have pushed a change that allows to unset properties by using `null`, e.g. ` cell.metadata.with({executionOrder: null});` resets the executionOrder property to its default value (which is `undefined` in this case)